### PR TITLE
Containers: retry for docker|podman build commands

### PR DESCRIPTION
**Pull request by jlausuch**
- Related tickets:
https://progress.opensuse.org/issues/107806
https://progress.opensuse.org/issues/107062

- Verification run: https://openqa.suse.de/tests/8300413

### CODE REVIEW

File: lib/containers/engine.pm [hunk 1]
```diff
@@ -109,9 +109,23 @@ sub build {
     my ($self, $dockerfile_path, $container_tag, %args) = @_;
     die 'wrong number of arguments' if @_ < 3;
 
-    #TODO add build with URL https://docs.docker.com/engine/reference/commandline/build/
-    $self->_engine_assert_script_run("build -f $dockerfile_path/Dockerfile -t $container_tag $dockerfile_path", $args{timeout} // 300);
```
Comment (pdostal) [inline lib/containers/engine.pm:113]: What about `script_retry`?

Comment (jlausuch) [reply lib/containers/engine.pm:113]: I think we need to do a cleanup of any possible intermediate image layer that have been created.

Comment (jlausuch) [reply lib/containers/engine.pm:113]: ok, will try that, thanks for the suggestion :)
But I'll wait until https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/14452 is merged and rebase on top.

**Comment by jlausuch** [comment]
Another failure today which could be avoided by this PR: 
https://openqa.suse.de/tests/8305339#step/image_docker/771

VR: http://openqa.suse.de/t8306388

**Review by pdostal — APPROVED** [review]
Nice! :smile:

**Pull request merged**